CSharpStepTemplate.yml

```diff
@@ -0,0 +1,142 @@
+# STEPS LEVEL TEMPLATE:
+# Used to build CSharp
+# Reusable
+# Used to "hide" the additional variables specific to this SDK which shouldn't be set from a higher level, or
+#   shared from a multi-build pipeline like a publish
+
+parameters:
+- name: ApiSpecSource
+  displayName: ApiSpecSource
+  type: string
+  default: -apiSpecGitUrl https://raw.githubusercontent.com/PlayFab/API_Specs/master/
+- name: CommitMessage
+  displayName: CommitMessage
+  type: string
+  default: Automated build from ADO Pipeline
+- name: GitDestBranch
+  displayName: GitDestBranch
+  type: string
+  default: doNotCommit
+- name: SdkName
+  displayName: SdkName
+  type: string
+  default: CSharpSDK
+- name: GitJSetupBranch
+  displayName: GitJSetupBranch
+  type: string
+  default: master
+- name: GitSdkGenBranch
+  displayName: GitSdkGenBranch
+  type: string
+  default: master
+- name: isVersioned
+  displayName: isVersioned
+  type: boolean
+  default: false
+- name: TestCompile
+  displayName: TestCompile
+  type: boolean
+  default: true
+- name: TestWin32NetCore
+  displayName: TestWin32NetCore
+  type: boolean
+  default: true
+- name: TestWin32Basic
+  displayName: TestWin32Basic
+  type: boolean
+  default: true
+- name: SelfTemplateResource
+  displayName: SelfTemplateResource
+  type: string
+  default: self
+
+steps:
+- checkout: JenkinsSdkSetupScripts
+  clean: true
+  path: s
+- checkout: API_Specs
+  clean: true
+  path: s/API_Specs
+- checkout: SdkGenerator
+  clean: true
+  path: s/SdkGenerator
+- checkout: CSharpSdk
+  clean: true
+  submodules: true
+  path: s/sdks/CSharpSDK
+  persistCredentials: true
+- bash: |
+    set -e
+    
+    echo alias the ADO variables into local variables
+    ApiSpecSource="${{ parameters.ApiSpecSource }}"
+    CommitMessage="${{ parameters.CommitMessage }}"
+    GitDestBranch="${{ parameters.GitDestBranch }}"
+    SdkName="${{ parameters.SdkName }}"
+    TestCompile="${{ parameters.TestCompile }}"
+    TestWin32Basic="${{ parameters.TestWin32Basic }}"
+    TestWin32NetCore="${{ parameters.TestWin32NetCore }}"
+    WORKSPACE=$(pwd -W)
+    # Hack attempt to get WORKSPACE into a sub-environment
+    export WORKSPACE="$WORKSPACE"
+    
+    echo === load util.sh to find msbuild ===
+    if [ -f "util.sh" ]; then
+      . "./util.sh" 2> /dev/null
+    elif [ ! -z "$WORKSPACE" ]; then
+      . "$WORKSPACE/JenkinsSdkSetupScripts/JenkinsScripts/Pipeline/util.sh" 2> /dev/null
+    fi
+    
+    Find2019MsBuild || Find2017MsBuild
+    
+    cd "$WORKSPACE/sdks/$SdkName"
+    
+    . "$WORKSPACE/JenkinsSdkSetupScripts/JenkinsScripts/Pipeline/testInit.sh"
+    
+    cd "$WORKSPACE/SDKGenerator/SDKBuildScripts"
+    targetSrc="csharp"
+    export PF_TEST_TITLE_DATA_JSON="$WORKSPACE\JenkinsSdkSetupScripts\Creds\testTitleData.json"
+    . ./shared_build.sh
+
```

Comment:
```suggestion
    
```
I guess it's not necessary, but Notepad++ yml parsing implies that the multiline string is broken by a blank-line that doesn't match the tabbing.